extras/rya.geoindexing/src/main/java/org/apache/rya/indexing/geotemporal/GeoTemporalIndexSetProvider.java

```diff
@@ -0,0 +1,237 @@
+        Optional<Collection<IndexingExpr>> geoFilters = Optional.empty();
+        Optional<Collection<IndexingExpr>> temporalFilters = Optional.empty();
+
+        //should only be 2 patterns.
+        for(final StatementPattern sp : patterns) {
+            final Var obj = sp.getObjectVar();
+
+            ///filter map does not have -const-
+
+
+            if(filterMap.containsKey(obj)) {
+                final Collection<IndexingExpr> filters = filterMap.get(obj);
+                final IndexingFunctionRegistry.FUNCTION_TYPE type = ensureSameType(filters);
+                if(type != null && type == FUNCTION_TYPE.GEO) {
+                    geoPattern = Optional.of(sp);
+                    geoFilters = Optional.of(filters);
+                    usedFilters.addAll(matchedFilters.get(obj));
+                } else if(type != null && type == FUNCTION_TYPE.TEMPORAL) {
+                    temporalPattern = Optional.of(sp);
+                    temporalFilters = Optional.of(filters);
+                    usedFilters.addAll(matchedFilters.get(obj));
+                } else {
+                    return null;
+                }
+            } else {
+                return null;
+            }
+        }
+
+        if(geoFilters.isPresent() && temporalFilters.isPresent() && geoPattern.isPresent() && temporalPattern.isPresent()) {
+            return new EventQueryNodeBuilder()
+                    .setStorage(eventStorage)
+                    .setGeoPattern(geoPattern.get())
+                    .setTemporalPattern(temporalPattern.get())
+                    .setGeoFilters(geoFilters.get())
+                    .setTemporalFilters(temporalFilters.get())
+                    .setUsedFilters(usedFilters)
+                    .build();
+        } else {
+            return null;
+        }
+    }
+
+    private FUNCTION_TYPE ensureSameType(final Collection<IndexingExpr> filters) {
+        FUNCTION_TYPE type = null;
+        for(final IndexingExpr filter : filters) {
+            if(type == null) {
+                type = IndexingFunctionRegistry.getFunctionType(filter.getFunction());
+            } else {
+                if(IndexingFunctionRegistry.getFunctionType(filter.getFunction()) != type) {
+                    return null;
+                }
+            }
+        }
+        return type;
+    }
+
+    private void buildMaps(final QuerySegment<EventQueryNode> node) {
+        final List<QueryModelNode> unused = new ArrayList<>();
+        for (final QueryModelNode pattern : node.getOrderedNodes()) {
+            if(pattern instanceof FunctionCall) {
+                discoverFilter((FunctionCall) pattern, unused);
+            }
+            if(pattern instanceof StatementPattern) {
+                discoverPatterns((StatementPattern) pattern, unused);
+            }
+        }
+    }
+
+    private void discoverFilter(final FunctionCall filter, final List<QueryModelNode> unmatched) {
```

Comment:
can't, I'd have to make the visitor static, which uses the maps in the class, and I can't make those static.